library/src/main/java/com/datatorrent/lib/fileaccess/FileAccess.java

```diff
@@ -151,7 +151,7 @@
      * @param value
      * @throws IOException
      */
-    void append(byte[] key, byte[] value) throws IOException;
+    void append(Slice key, Slice value) throws IOException;
```

Comment:
Keeping the old method and deprecating it seems a good choice when it does not cause extra complication and helps the downstream consumer.